api/src/main/java/com/velocitypowered/api/event/player/PlayerResourcePackSendEvent.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2018 Velocity Contributors
+ *
+ * The Velocity API is licensed under the terms of the MIT License. For more details,
+ * reference the LICENSE file in the api top-level directory.
+ */
+
+package com.velocitypowered.api.event.player;
+
+import com.velocitypowered.api.event.ResultedEvent;
+import net.kyori.adventure.text.Component;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/**
+ * This event is fired when the downstream server tries to send a player a ResourcePackSend packet.
+ * Velocity will wait on this event to finish before forwarding the packet to the user. If this
+ * event is denied, it will retroactively send a DENIED status to the downstream server in response.
+ * If the downstream server has it set to "forced" it will forcefully disconnect the user.
+ */
+public class PlayerResourcePackSendEvent implements ResultedEvent<ResultedEvent.GenericResult> {
```

Comment:
once we move to java 16+ if we do, we might want to just turn this into a record, it doesn't need to be a class realistically.